### PR TITLE
crds: Add proxy_url support for Probes

### PR DESCRIPTION
**Pull request by simonswine**
This passes through the proxy_url argument for probe CRDs

**Release Note Template (will be copied)**

```release-note:feature
Add proxy_url support for Probes
```

### CODE REVIEW

File: pkg/prometheus/promcfg.go [hunk 1]
```diff
@@ -709,6 +709,9 @@ func (cg *ConfigGenerator) generateProbeConfig(
 	if m.Spec.ProberSpec.Scheme != "" {
 		cfg = append(cfg, yaml.MapItem{Key: "scheme", Value: m.Spec.ProberSpec.Scheme})
 	}
+	if m.Spec.ProberSpec.ProxyURL != "" {
```
Comment (paulfantom) [inline pkg/prometheus/promcfg.go:712]: Could we have a unit test for this?

File: pkg/apis/monitoring/v1/types.go [hunk 2]
```diff
@@ -951,6 +951,8 @@ type ProberSpec struct {
 	// Path to collect metrics from.
 	// Defaults to `/probe`.
 	Path string `json:"path,omitempty"`
+	// Optional ProxyURL
```
Comment (simonpasquier) [inline pkg/apis/monitoring/v1/types.go:954]:
```suggestion
	// Optional ProxyURL.
```

Comment (simonswine) [reply pkg/apis/monitoring/v1/types.go]: Added in 9b7c9d9

File: pkg/prometheus/promcfg.go (hunk 1 above)
Comment (simonswine) [reply pkg/prometheus/promcfg.go:712]: I thought adding a separate method is overkills, so I just added the field in the existing test case: 185a4f3

**Review by simonpasquier — APPROVED** [review]
lgtm

**Pull request merged**